### PR TITLE
Feature/fetch pre trained model list

### PR DESCRIPTION
**Pull request by wanglam**
### Description
fetch pre trained model data for register model type modal and register model form.

### Issues Resolved
_List any issues this PR will resolve, e.g. Closes [...]._ 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: public/components/register_model/register_model.tsx [hunk 1]
```diff
@@ -179,6 +182,22 @@ export const RegisterModelForm = () => {
...
+    const subscriber = modelRepositoryManager
+      .getPreTrainedModel$(nameParams, 'torch_script')
+      .subscribe((preTrainedModel) => {
+        // TODO: store pre-trained model data
+        // eslint-disable-next-line no-console
+        console.log(preTrainedModel);
+      });
+    return () => {
+      subscriber.unsubscribe();
+    };
```
Comment (ruanyl) [inline public/components/register_model/register_model.tsx:198]: Ideally, `useEffect` should return cleanup function, what about change it a bit like this?
```suggestion
    function subscribeToModel(name: string | null) {
      if (!name) {
        return;
      }
      return modelRepositoryManager
        .getPreTrainedModel$(name, 'torch_script')
        .subscribe((preTrainedModel) => {
          // TODO: store pre-trained model data
          // eslint-disable-next-line no-console
          console.log(preTrainedModel);
        });
    }
    const subscriber = subscribeToModel(nameParams);
    return () => {
      if (subscriber) {
        subscriber.unsubscribe();
      }
    };
```

File: server/routes/model_repository_router.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  MODEL_REPOSITORY_CONFIG_URL_API_ENDPOINT,
+} from './constants';
+
+const fetchURLAsJSONData = (url: string) => fetch(url).then((response: any) => response.json());
+
+export const modelRepositoryRouter = (router: IRouter) => {
+  router.get({ path: MODEL_REPOSITORY_API_ENDPOINT, validate: false }, async () => {
+    try {
+      const data = await fetchURLAsJSONData(
+        'https://raw.githubusercontent.com/opensearch-project/ml-commons/2.x/ml-algorithms/src/test/resources/org/opensearch/ml/engine/algorithms/text_embedding/pre_trained_models_list.json'
```
[the remark this answers is not in the document]
Comment (wanglam) [reply server/routes/model_repository_router.ts:22]: We will changed to use the main branch file after merged to the main branch. I can double confirm with @dhrubo-os about this link address for production using.

[the remark this answers is not in the document]
Comment (dhrubo-os) [reply server/routes/model_repository_router.ts:22]: I believe this file will be there. We will also merge this commit with main branch in near future. I would say that will be the most stable link. But for now, we can use this link. 

@ylwu-amzn what do you think?

[the remark this answers is not in the document]
Comment (dhrubo-os) [reply server/routes/model_repository_router.ts:22]: We merged this with mainline already. I think we can use this link:

https://github.com/opensearch-project/ml-commons/blob/main/ml-algorithms/src/test/resources/org/opensearch/ml/engine/algorithms/text_embedding/pre_trained_models_list.json

[the remark this answers is not in the document]
Comment (ylwu-amzn) [reply server/routes/model_repository_router.ts:22]: I think using main branch link should be good

File: public/components/register_model/register_model.tsx (hunk 1 above)
Comment (wanglam) [reply public/components/register_model/register_model.tsx:198]: Are there any rules about this part? The code you suggested would be a bit complicated. I think the `useEffect` accept two forms of return value. If the name parameter exists, we will return the clean-up function which means will clean the subscribe when name changed. If it's not exists, we don't subscribe anything, we won't provide the clean-up function.

**Review by ruanyl — APPROVED** [review]

**Pull request merged**